widget/form.go

```diff
@@ -241,6 +295,39 @@ func (f *Form) setUpValidation(widget fyne.CanvasObject, i int) {
 		}
 		w.SetOnValidationChanged(updateValidation)
 	}
+	// setup triggers to call f.Validate whenever the widget changes state
+	if e, ok := widget.(*Entry); ok {
+		e.OnChanged = func(string) {
+			f.Validate()
+		}
+		return
+	}
+	if c, ok := widget.(*Check); ok {
+		c.OnChanged = func(bool) {
+			f.Validate()
+		}
+		return
+	}
+	if cg, ok := widget.(*CheckGroup); ok {
+		cg.OnChanged = func([]string) {
+			f.Validate()
+		}
+	}
+	if s, ok := widget.(*Select); ok {
+		s.OnChanged = func(string) {
+			f.Validate()
+		}
+	}
+	if se, ok := widget.(*SelectEntry); ok {
+		se.OnChanged = func(string) {
+			f.Validate()
+		}
+	}
+	if s, ok := widget.(*Slider); ok {
+		s.OnChanged = func(float64) {
+			f.Validate()
+		}
+	}
```

Comment:
Sorry, but I don't think this is the right way to tackle the problem. Not only are you overwriting `OnChanged` which may have been set by the user, but you are also hooking translations into (and hard-coding for) widgets that don't currently support validation.

Reply:
Perhaps we could resolve the problem of overwriting `OnChanged`, by calling the original func that was there after we called `f.Validate()`. The user could still overwrite our function though, not sure how much of a problem that would be.
As I laid out in my other answer hooking into the validation change event isn't enough, and as far as I can tell there is no uniform way to hook into the `OnChanged` event of a widget since they all have a different signature for their `OnChanged` functions. Entries use `func(string)` for example while check boxes use `func(bool)`.
Hardcoding widgets and their `OnChanged` functions isn't elegant but I don't know how to avoid that.
I also wrote why checkboxes need to be involved at all (even if they don't support validation) in my second answer.

Reply:
Hard-coding widgets is going to be viable in the long run. Any develop has the ability to create custom widgets and we can't keep track of those as well. This might require some more though to get right. I understand that it is a tricky problem to solve.

Reply:
Instead of updating on every widget change, we could leave that up to the developer. They would need to set the `OnChanged` function of very widget the `Validator` would check to include a call to `f.Validate()`. This would also be beneficial to the performance since the `Validator` would only be run if a widget, the validator actually cares about, changes.

Reply:
> Instead of updating on every widget change, we could leave that up to the developer. 

It sounds quite unintuitive which isn't really our approach.
Setting on a validator for Form should just work, and update at the right time.
If we can detect `Validatable` instead it would be much cleaner and extensible.